google-cloud-storage/src/main/java/com/google/cloud/storage/Notification.java

```diff
@@ -0,0 +1,313 @@
+import java.util.Objects;
+
+/**
+ * Google Storage Notification metadata;
+ *
+ * @see <a href="https://cloud.google.com/storage/docs/concepts-techniques#concepts">Concepts and
+ *     Terminology</a>
+ */
+public class Notification implements Serializable {
+
+  private static final long serialVersionUID = 5725883368559753810L;
+  private static final PathTemplate PATH_TEMPLATE =
+      PathTemplate.createWithoutUrlEncoding("projects/{project}/topics/{topic}");
+
+  public enum PayloadFormat {
+    JSON_API_V1,
+    NONE
+  }
+
+  static final Function<com.google.api.services.storage.model.Notification, Notification>
+      FROM_PB_FUNCTION =
+          new Function<com.google.api.services.storage.model.Notification, Notification>() {
+            @Override
+            public Notification apply(com.google.api.services.storage.model.Notification pb) {
+              return Notification.fromPb(pb);
+            }
+          };
+  static final Function<Notification, com.google.api.services.storage.model.Notification>
+      TO_PB_FUNCTION =
+          new Function<Notification, com.google.api.services.storage.model.Notification>() {
+            @Override
+            public com.google.api.services.storage.model.Notification apply(
+                Notification notification) {
+              return notification.toPb();
+            }
+          };
+  private final String generatedId;
+  private final String topic;
+  private final List<String> eventTypes;
+  private final Map<String, String> customAttributes;
+  private final PayloadFormat payloadFormat;
+  private final String objectNamePrefix;
+  private final String etag;
+  private final String selfLink;
+
+  /** Builder for {@code NotificatioInfo}. */
+  public static class Builder {
+
+    private String generatedId;
+    private String topic;
+    private List<String> eventTypes;
+    private Map<String, String> customAttributes;
+    private PayloadFormat payloadFormat;
+    private String objectNamePrefix;
+    private String etag;
+    private String selfLink;
+
+    private Builder(String topic) {
+      this.topic = topic;
+    }
+
+    private Builder(Notification notification) {
+      generatedId = notification.generatedId;
+      etag = notification.etag;
+      selfLink = notification.selfLink;
+      topic = notification.topic;
+      eventTypes = notification.eventTypes;
+      customAttributes = notification.customAttributes;
+      payloadFormat = notification.payloadFormat;
+      objectNamePrefix = notification.objectNamePrefix;
+    }
+
+    public Builder setGeneratedId(String generatedId) {
+      this.generatedId = generatedId;
+      return this;
+    }
+
+    public Builder setSelfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return this;
+    }
+
+    /** The name of the topic. It must have the format "projects/{project}/topics/{topic}". */
+    public Builder setTopic(String topic) {
+      this.topic = topic;
+      return this;
+    }
```

Comment:
You need to to validate that the given topic meets the format

Reply:
yes. we can validate in` build()` instead of here.